server/services/DashboardService/DashboardService.test.js

```diff
@@ -0,0 +1,208 @@
+import moment from 'moment'
+import DashboardService from '.'
+import {
+  createDb,
+  createAnalysisConfig,
+  createSubjectDayData,
+} from '../../../test/fixtures'
+
+const db = createDb()
+const config = [
+  createAnalysisConfig(),
+  createAnalysisConfig({
+    label: 'Jump',
+    analysis: 'jump_of',
+    variable: 'jumpVariable',
+    category: 'power',
+  }),
+  createAnalysisConfig({
+    label: 'Read',
+    analysis: 'read_of',
+    variable: 'readVariable',
+    category: 'reading',
+  }),
+]
+const study = 'YA'
+const subject = 'YA01'
+const sizeVariableData = [
+  createSubjectDayData({
+    day: 1,
+    sizeVariable: 30,
+  }),
+  createSubjectDayData({
+    day: 45,
+    sizeVariable: 2,
+  }),
+]
+const jumpVariableData = [
+  createSubjectDayData({
+    day: 10,
+    jumpVariable: 1,
+  }),
+  createSubjectDayData({
+    day: 20,
+    jumpVariable: 30,
+  }),
+]
+const readVariableData = [
+  createSubjectDayData({
+    day: 30,
+    readVariable: 3,
+  }),
+  createSubjectDayData({
+    day: 32,
+    readVariable: 4,
+  }),
+]
+
+describe(DashboardService, () => {
+  const dService = new DashboardService(db, study, subject, config)
+
+  describe(DashboardService._assessmentsFromConfig, () => {
```

Comment:
we shouldn't be testing private methods, only public ones. If we want to get this additional test coverage it is a sign that we should split this service into multiple parts (modules/classes/etc) that can be tested individually.

Reply:
6b8e25afce0ca081cd421f151aad983920c509ba